File: 프로그래머스/unrated/181887. 홀수 vs 짝수/홀수 vs 짝수.js
function solution(num_list) {
    const oddSum = num_list.filter((_, index) => index % 2 === 1).reduce((a, b) => a + b, 0);
  const evenSum = num_list.filter((_, index) => index % 2 === 0).reduce((a, b) => a + b, 0);
  
  return Math.max(oddSum, evenSum);
}

File: 프로그래머스/unrated/181887. 홀수 vs 짝수/홀수 vs 짝수.java
class Solution {
    public int solution(int[] num_list) {
        int odd = 0;
        int even = 0;
        for(int i=0; i<num_list.length; i++) {
            if(i % 2 == 0) even += num_list[i];
            else if(i % 2 == 1) odd += num_list[i];
            else return num_list[i];
        }
        return odd > even ? odd : even;
    }
}